expression/builtin_cast.go

```diff
@@ -76,6 +78,192 @@ func (b *builtinCastSig) eval(row []types.Datum) (d types.Datum, err error) {
 	return d, errors.Errorf("unknown cast type - %v", b.tp)
 }
 
+type castAsIntFunctionClass struct {
```

Comment:
Add some comment for the following types. For example:
We implement castAsXXFunctionClass for .....